application/tests/mock/phone/Component.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace tests\mock\phone;
+
+use Sil\IdpPw\Common\PhoneVerification\PhoneVerificationInterface;
+use Sil\IdpPw\Common\PhoneVerification\NotMatchException;
+use yii\base\Component as YiiComponent;
+
+class Component extends YiiComponent implements PhoneVerificationInterface
+{
+
+    public $codeLength;
+
+    /**
+     * Initiate phone verification, returns $code or newly generated code/identifier to be stored in Reset model
+     * @param string $phoneNumber
+     * @param string $code
+     * @return string The verification code used, or another identifier to be used with self::verify() later
+     * @throws \Exception
+     */
+    public function send($phoneNumber, $code)
+    {
+
+        // Look up code by phone number to support "generated" numbers
+        $data = include __DIR__ . '/data.php';
+        foreach ($data as $phone) {
+            if ($phone['number'] == $phoneNumber) {
+                return $phone['code'];
+            }
+        }
+
+        return $code;
+    }
+
+    /**
+     * Verify that previously stored $resetCode matches the code provided by the user, $userProvided
+     * Component may use $resetCode as a key for it's own purposes, as is the case with the Nexmo Verify service.
+     * Return true on success or throw NotMatchException when match fails.
+     * Throw \Exception when other exception occurs like network issue with service provider
+     * @param string $resetCode
+     * @param string $userProvided
+     * @return boolean
+     * @throws \Exception
+     * @throws \Sil\IdpPw\Common\PhoneVerification\NotMatchException
+     */
+    public function verify($resetCode, $userProvided)
```

Comment:
What if we checked the `$resetCode` against the dummy `id` values first. If it matched an `id`, then we go ahead and return true or throw a `NotMatchException` based on whether `$userProvided` matches the value in that dummy data entry. If `$resetCode` didn't match any `id` values, then we could return/throw based on whether `$resetCode` matched `$userProvided`. That should at least avoid some false positives, though it still seems like there's probably some possibility for a wrong response if someone really tried to get one (which wouldn't make sense, since this is just a mock for tests).